include/dfu/mcuboot.h

```diff
@@ -181,6 +220,17 @@ int boot_request_upgrade(int permanent);
  */
 int boot_erase_img_bank(uint8_t area_id);
 
+/**
+ * @brief Read the image swap state
+ *
+ * @param flash_area_id Id of flash parttition from which trailer will be read.
+ * @param state Struture for holding swap state.
+ *
+ * @return 0 on success, negative errno code on fail.
+ */
+int mcuboot_read_swap_state_by_id(int flash_area_id,
+				  struct mcuboot_swap_state *state);
```

Comment:
Is there a rationale for adding this function and `struct mcuboot_swap_state` instead of just using `boot_read_swap_state_by_id` and `struct boot_swap_state`?

Reply:
boot_read_swap_state_by_id() can't be made zephyr API. It can only be used by module which made a module shim (as it links with MCUBOOT_BOOTUTIL) (by convention). For the MCUBoot shell  i can link it with the library as well, and left behind this API.
Do you think that this API will be rather not useful for other?

Reply:
@utzig ^^?

Reply:
Adding `image_num` would be a short term fix. We should not blindly cast one structure to the other, because that is error prone. Somebody will change order of fields or add new information to `struct boot_swap_state` and we will be screwed once again.

So we should either:
- add helper `struct boot_swap_state` variable and copy each structure field separately,
- link `MCUBOOT_BOOTUTIL` to mcuboot shell or other modules that use mcuboot public API.

Reply:
> - link MCUBOOT_BOOTUTIL to mcuboot shell or other modules that use mcuboot public API.

Ok, I will go with that

Reply:
done